### ISSUE TITLE
Toast pop up randomly when other popup/dialog opens

### ISSUE THREAD

**Issue by Shubham-1994** [calintamas/react-native-toast-message#428]
**Describe the bug**
I have declared Toast component in the Root Container. Because of this while any other dialog or popup like Android Permission dialog appears on screen, an Empty toast also shows on top of screen.

**Expected behavior**
Toast should not be displayed until Toast.show is called

**Environment (please complete the following information):**
 - OS: Android
 - react-native-toast-message version: ^2.1.5
 - react-native version 0.67.2

**Comment by casq10pearls** [comment]
The same issue here.
could you please answer us something?

**Comment by vatsalya-honeycomb** [comment]
I am also able to reproduce it in Android, IOS works fine.

**Comment by ihsan77** [comment]
@vatsalya-honeycomb is there any solution for android?

**Comment by ihsan77** [comment]
@calintamas any solution for android?

**Comment by casq10pearls** [comment]
Thanks @Shubham-1994,  the second option works!

**Comment by heinst** [comment]
> One more hack is if you are not using default success toast variant then override it with null.

In your ToastConfig add 

`success: props => null,`

**Comment by nlaffey** [comment]
The fix noted by @elabayoub doesn't seem to be compatible with version 2.1.5 unfortunately. The index.js file is different and I can't seem to quickly find the equivalent change.

**Comment by mehranjavid** [comment]
I was having the same issue this worked for me
https://github.com/calintamas/react-native-toast-message/issues/226#issuecomment-1009841777

**Comment by rchavik** [comment]
> The fix noted by @elabayoub doesn't seem to be compatible with version 2.1.5 unfortunately. The index.js file is different and I can't seem to quickly find the equivalent change.

How about this?

```diff
diff --git a/node_modules/react-native-toast-message/lib/src/ToastUI.js b/node_modules/react-native-toast-message/lib/src/ToastUI.js
index bb5ce5c..2720e98 100644
--- a/node_modules/react-native-toast-message/lib/src/ToastUI.js
+++ b/node_modules/react-native-toast-message/lib/src/ToastUI.js
@@ -34,6 +34,7 @@ function renderComponent({ data, options, config, isVisible, show, hide }) {
 export function ToastUI(props) {
     const { isVisible, options, hide } = props;
     const { position, topOffset, bottomOffset, keyboardOffset } = options;
+    if (!isVisible) return (<></>);
     return (<AnimatedContainer isVisible={isVisible} position={position} topOffset={topOffset} bottomOffset={bottomOffset} keyboardOffset={keyboardOffset} onHide={hide}>
       {renderComponent(props)}
     </AnimatedContainer>);
```

**Comment by rchavik** [comment]
> > > The fix noted by @elabayoub doesn't seem to be compatible with version 2.1.5 unfortunately. The index.js file is different and I can't seem to quickly find the equivalent change.
> > 
> > 
> > How about this?
> > ```diff
> > diff --git a/node_modules/react-native-toast-message/lib/src/ToastUI.js b/node_modules/react-native-toast-message/lib/src/ToastUI.js
> > index bb5ce5c..2720e98 100644
> > --- a/node_modules/react-native-toast-message/lib/src/ToastUI.js
> > +++ b/node_modules/react-native-toast-message/lib/src/ToastUI.js
> > @@ -34,6 +34,7 @@ function renderComponent({ data, options, config, isVisible, show, hide }) {
> >  export function ToastUI(props) {
> >      const { isVisible, options, hide } = props;
> >      const { position, topOffset, bottomOffset, keyboardOffset } = options;
> > +    if (!isVisible) return (<></>);
> >      return (<AnimatedContainer isVisible={isVisible} position={position} topOffset={topOffset} bottomOffset={bottomOffset} keyboardOffset={keyboardOffset} onHide={hide}>
> >        {renderComponent(props)}
> >      </AnimatedContainer>);
> > ```
> 
> Thanks this patch worked for me. Although one thing I noticed is now the toast doesn't have the slide off animation, it just disappears. Not a big problem but the slide on animation still works fine 🧐.. Any ideas why?

Because the component that does the animation does not get rendered at all. So instead of being able to slide offscreen, you just get an empty fragment.

**Comment by Ali-Aref** [comment]
> > The fix noted by @elabayoub doesn't seem to be compatible with version 2.1.5 unfortunately. The index.js file is different and I can't seem to quickly find the equivalent change.
> 
> How about this?
> 
> ```diff
> diff --git a/node_modules/react-native-toast-message/lib/src/ToastUI.js b/node_modules/react-native-toast-message/lib/src/ToastUI.js
> index bb5ce5c..2720e98 100644
> --- a/node_modules/react-native-toast-message/lib/src/ToastUI.js
> +++ b/node_modules/react-native-toast-message/lib/src/ToastUI.js
> @@ -34,6 +34,7 @@ function renderComponent({ data, options, config, isVisible, show, hide }) {
>  export function ToastUI(props) {
>      const { isVisible, options, hide } = props;
>      const { position, topOffset, bottomOffset, keyboardOffset } = options;
> +    if (!isVisible) return (<></>);
>      return (<AnimatedContainer isVisible={isVisible} position={position} topOffset={topOffset} bottomOffset={bottomOffset} keyboardOffset={keyboardOffset} onHide={hide}>
>        {renderComponent(props)}
>      </AnimatedContainer>);
> ```

Thanks this worked for me.

**Comment by calintamas** [comment]
The workaround to not use the default `success` type should work indeed. 

The reason behind this entire thing is an Android-only issue with `Animated` https://github.com/facebook/react-native/issues/28574

I'll disable `useNativeDriver` on Android. It should fix it I think.

**Comment by calintamas** [comment]
Released in `2.2.1`. 

Please let me know if there are any issues, thanks!
